### DISCUSSION TITLE
how to aceed json datas or collections from javascript with eleventy ?

### DISCUSSION THREAD

**Discussion by Gaelle-Bono** [11ty/eleventy#3287] [Q&A]
Hi, 
I'm a very beginner in eleventy. 
I'm sorry also for my english. 
In a eleventy project,  I have a js file called gallery.js in src/js,
In this file i 've got a listener that listen the click on a dropdown. 
If an item (a photo category) has been selected, i'm generating an html part specific (photos list corresponding to the selected category in the dropdown menu)  that i insert in a div.  
If all the datas are in my javascript, it is working. 
But i would like to test two other processes. 

If the datas are present in src/_data/photos.json I don't know how to acceed to these datas from my javascript file. 
Second process. 
I created a collection of portraits photos for exemple in .eleventy.js named portraits. 
If i call collections.portraits from html, it is working and i can see my portraits photos with a loop. 
But 
from javascript, i don't know how to acceed to collections.portraits created in .eleventy.js.

 can you help me ? 

Gaëlle

**Comment by MyXoToD** [comment]
As far as I know this is not possible. 11ty will build the site and output only static files (not including the `.eleventy.js` file at runtime). So you can not access it with a JavaScript file after build time. Depending on what you specifically want to do, you could add all the data to your HTML (which seems to already work for you) and hide the content by default that is not needed. Use your custom JavaScript to show/hide the requested content at run time. Let me know if you need any more help.

**Comment by Gaelle-Bono** [comment]
Thank you very much for your answer ! 
I understood now how to do it.
